gbpservice/neutron/nsf/configurator/modules/firewall.py

```diff
@@ -0,0 +1,347 @@
+                LOG.error(msg)
+                raise Exception(msg)
+        except Exception as err:
+            LOG.error("Failed to perform the operation: %s. %s"
+                      % (ev.id, str(err).capitalize()))
+        finally:
+            self._sc.event_done(ev)
+
+
+def _create_rpc_agent(sc, topic, manager):
+    return RpcAgent(sc,
+                    host=cfg.CONF.host,
+                    topic,
+                    manager)
+
+
+def rpc_init(sc, conf):
+    fwrpcmgr = FwaasRpcReceiver(conf, sc)
+    vyosfwrpcmgr = FwGenericConfigRpcReceiver(conf, sc)
+
+    fwagent = _create_rpc_agent(sc, const.FIREWALL_RPC_TOPIC, fwrpcmgr)
+    vyosfwagent = _create_rpc_agent(sc, const.VYOS_FIREWALL_RPC_TOPIC,
+                                    vyosfwrpcmgr)
+
+    sc.register_rpc_agents([fwagent, vyosfwagent])
+
+
+def events_init(sc, drivers):
+    evs = [
+        Event(id='CONFIGURE_FW', handler=FwaasHandler(sc, drivers)),
+        Event(id='UPDATE_FW', handler=FwaasHandler(sc, drivers)),
+        Event(id='DELETE_FW', handler=FwaasHandler(sc, drivers)),
+
+        Event(id='CONFIGURE_INTERFACES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CLEAR_INTERFACES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CONFIGURE_LICENSE', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='RELEASE_LICENSE', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CONFIGURE_SOURCE_ROUTES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='DELETE_SOURCE_ROUTES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='ADD_PERSISTENT_RULE', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='DEL_PERSISTENT_RULE', handler=FwGenericConfigHandler(
+                                                                sc, drivers))]
+    sc.register_events(evs)
+
+
+def load_drivers():
+    ''' Create object of firewall drivers
+    '''
+    drivers = {"vyos_fwaas": 'FwaasDriver',
+               "vyos_config": 'FwGenericConfigDriver'}
+    return drivers
+
+
+def module_init(sc, conf):
+    drivers = load_drivers()
+    try:
+        events_init(sc, drivers)
+    except Exception as err:
+        LOG.error("Events initialization unsuccessful. %s"
+                  % (str(err).capitalize()))
+        raise err
+    else:
+        LOG.debug("Events initialization successful.")
+
+    msg = ("RPC topics are: %s and %s."
+           % (const.FIREWALL_RPC_TOPIC,
+              const.VYOS_FIREWALL_RPC_TOPIC))
```

Comment:
Why is this topic used? Is this for generic configuration? We should not have vendor specific code in modules.